### ISSUE TITLE
USWDS - Combobox & Timepicker: Verbose and unclear screen reader announcements make selecting and clearing a time difficult for some users

### ISSUE THREAD

**Issue by jaclinec** [uswds/uswds#5905]
# Summary
The instructions announced for screen readers when a user focuses on the time picker are overly verbose and confusing, which could cause some people to be so confused that they fail to select a time. 

Additionally, the announcement to screen readers for clearing the time (i.e. "clear select contents") is unclear. 

# Observations

- 3 blind participants (2 using JAWS, 1 using NVDA) found the instructions announced excessive and unhelpful in guiding them to selecting a time. One of those participants had trouble selecting a time because she didn't realize she had to hit 'enter' to select a time. The instruction to select 'enter' was buried in other verbose instructions. See [video clips](https://drive.google.com/file/d/1OYwhaIaFVHllUcUiEWrMi9pviMln0ROf/view?usp=sharing) :lock:

> “it causes confusion with the information and…I have to listen [to] all of those carefully.”

- 1 blind user was confused by the 'clear select contents' announcement and said that it would be clearer to him if it announced something more specific like "clear the time."

# Affected user groups

- Screen reader users

# Research method
Usability testing with 8 participants with disabilities: 
- 5 visually impaired (3 blind, 2 low vision/legally blind (1 with nystagmus))
- 1 motor impairment
- 1 ADHD
- 1 anxiety and depression

Screen reader software participants used: JAWS, NVDA, and VoiceOver

[Link to public findings report forthcoming...here is the internal [Zebra findings report](https://docs.google.com/document/d/1vbGSh4UNZiseyzdGXa1pJwU_3M4y7HaU3QPkXOa7gsI/edit?usp=sharing)] :lock:

# Next steps

- [ ] Adjust the instructions read by screen readers to be shorter, clearer, and more helpful. 
- [ ] Change the ‘clear contents’ announcement to ‘clear your time selection’ or similar.

**Comment by mejiaj** [comment]
Instructions are coming from combobox:

https://github.com/uswds/uswds/blob/a1052196378b62475c7dea621f262d0fbcc7f318/packages/usa-combo-box/src/index.js#L264-L267

**Comment by amycole501** [comment]
Retesting this; James is correct, it's the combo box element reading a lot of that out loud. However a separate issue is the fact that the time picker label of hh:mm is NOT read aloud for someone to type into it properly. Also it doesn't give you a notification that the times are only in half hour increments.

**Comment by amyleadem** [comment]
Update: 
1. **The "excessive and unhelpful" instructions should be resolved with the release of [3.9.0](https://github.com/uswds/uswds/releases/tag/v3.9.0) (PR #6022)**. 
    1. @jaclinec, @amycole501, or @alex-hull, can you confirm that this part of the issue has been resolved? 
2. **The screen reader still reads "Clear the select contents" when focusing on the "X" icon.** 
    1. Developer's note: this content comes from an [aria-label hard-coded into the combo box](https://github.com/uswds/uswds/blob/v3.9.0/packages/usa-combo-box/src/index.js#L247). We would need to build in the capability to customize that label so that we can change it in time picker.

@jaclinec, if the only issue remaining is related to the "Clear the select contents" readout, does the weight/priority of the issue change?

**Comment by jaclinec** [comment]
@amyleadem The most critical issue to resolve was the first issue about verbose instructions since that impacted the most participants negatively. 

The second issue for "clear the select contents" is a lower priority since only 1 participant seemed to have an issue with it, although I think it makes sense to make the instructions clearer (and it seems like a low lift improvement). So on one hand, I think it'd be great to just wrap up all the improvements for this component so maybe we _should_ prioritize it. On the other hand, considering developers' workloads and other priorities, I think if we _have_ to de-prioritize it, that's ok.

**Comment by amyleadem** [comment]
To do:
- Create a sub-issue for the "clear select contents" piece of this issue
- Explore "Clear selection" as an alternative label 
- Explore removing hard-coded label, allow for easy override of content

**Comment by amyleadem** [comment]
Removing this from the November release because we need to do some usability research before moving forward with issue #6151

**Comment by chandracarney** [comment]
In June, we discussed that this needs further testing and triage with [#6528](https://github.com/uswds/uswds/issues/6528). @jonathanbobel did a little digging again recently, so I'll un-assign myself and assign Jon for the time being.

**Comment by ryparker** [comment]
I'm closing the separate research task #6150 and keeping the work here. We still need to test the remaining time-picker announcements and clear-selection wording with affected users, using the findings above and coordinating with #6151 and #6528.

This issue stays open. The earlier instruction change doesn't resolve all of the reported usability concerns.
